### PR TITLE
feat: add wait for masternode-status "ready"

### PR DESCRIPTION
**Pull request by shotonoff**
## Issue being fixed or feature implemented

In some cases Tenderdash may crash if it runs in `validator` mode and it can't get pro-tx hash by some external reason.
As result, Tenderdash crashes and docker supervisor runs it again until node gets pro-tx hash

## What was done?

Implement wait-for logic, that checks the `masternode status` until get `READY`. If the status is `WAITING_FOR_PROTX` then waits for 1 second and retry the operation, otherwise exit.

## How Has This Been Tested?

The function covers with the test

## Breaking Changes

N/A

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: dash/core/client.go [hunk 1]
```diff
@@ -220,3 +225,26 @@ func (rpcClient *RPCClient) QuorumVerify(
...
+	for {
+		result, err := client.MasternodeStatus()
+		if err != nil {
+			return fmt.Errorf("failed to get masternode status: %w", err)
+		}
+		switch result.State {
+		case btcjson.MNStatusStateReady:
+			return nil
+		case btcjson.MNStatusStateWaitingForProtx:
+			sleep(retryTimeout)
```
[the remark this answers is not in the document]
Comment (shotonoff) [reply dash/core/client.go:240]: this place can generate too many logs, so instead I added the log before and after calling `WaitForMNReady`

File: dash/core/client.go [hunk 2]
```diff
@@ -13,6 +16,8 @@ import (
 
 const ModuleName = "rpcclient"
 
+var sleep = time.Sleep
```
[the remark this answers is not in the document]
Comment (shotonoff) [reply dash/core/client.go:19]: to be able to override `sleep` function in a test

**Review by shotonoff — COMMENTED** [review]
made modifications, please have a look at them

**Review by shotonoff — COMMENTED** [review]
answered to the comment

**Review by shotonoff — COMMENTED** [review]
answered

**Review by lklimek — APPROVED** [review]

**Pull request merged**